pom.xml

```diff
@@ -129,6 +129,28 @@
                 </pluginManagement>
             </build>
         </profile>
+        <profile>
+            <id>sonar-cloud</id>
+            <properties>
+                <!-- SonarQube rule exclusions for: Integration tests -->
+                <sonar.issue.ignore.multicriteria>e1,e2,e3,e4,e5</sonar.issue.ignore.multicriteria>
+                <!-- Disable: Public types, methods and fields (API) should be documented with Javadoc -->
+                <sonar.issue.ignore.multicriteria.e1.ruleKey>java:S1176</sonar.issue.ignore.multicriteria.e1.ruleKey>
+                <sonar.issue.ignore.multicriteria.e1.resourceKey>**/*-integration-tests/**/*.java</sonar.issue.ignore.multicriteria.e1.resourceKey>
+                <!-- Disable: Constructors should only call non-overridable methods -->
+                <sonar.issue.ignore.multicriteria.e2.ruleKey>java:S1699</sonar.issue.ignore.multicriteria.e2.ruleKey>
+                <sonar.issue.ignore.multicriteria.e2.resourceKey>**/*-integration-tests/**/*.java</sonar.issue.ignore.multicriteria.e2.resourceKey>
+                <!-- Disable: String literals should not be duplicated -->
+                <sonar.issue.ignore.multicriteria.e3.ruleKey>java:S1192</sonar.issue.ignore.multicriteria.e3.ruleKey>
+                <sonar.issue.ignore.multicriteria.e3.resourceKey>**/*-integration-tests/**/*.java</sonar.issue.ignore.multicriteria.e3.resourceKey>
+                <!-- Disable: Inheritance tree of classes should not be too deep -->
```

Comment:
Reported as code smell with high severity. We have some test base classes that exceed this limit, but I don't see a lot of potential for issues here.